### PR TITLE
sql: map correct OID for timestamptz datums

### LINKED ISSUE #7636: pq: unknown datum type: timestamptz
This code (the important line is the last one, but the whole thing can be used to reproduce)

```go
package main

import (
	"database/sql"
	"log"
	"time"

	_ "github.com/lib/pq"
)

func main() {
	db, _ := sql.Open("postgres", "user=root port=26257 sslmode=disable")
	db.Query(`CREATE DATABASE "timestamptz_test1"`)

	db, _ = sql.Open("postgres", "user=root port=26257 dbname=timestamptz_test1 sslmode=disable")
	db.Query(`CREATE TABLE "test1" ("sometime" timestamp with time zone)`)

	log.Println(db.Query(`INSERT INTO "test1" ("sometime") VALUES ($1)`, time.Now()))
}
```

gives me
```sh
2016/07/05 21:41:25 <nil> pq: unknown datum type: timestamptz
```

I'm using `beta-20160629`.

### PR DESCRIPTION
**Pull request by dt**
Fixes #7636

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7642)

### CODE REVIEW

File: sql/pgwire_test.go [hunk 1]
```diff
@@ -765,13 +765,13 @@ func TestPGPreparedExec(t *testing.T) {
...
 		{
-			"CREATE TABLE d.types (i int, f float, s string, b bytes, d date, m timestamp, n interval, o bool, e decimal)",
+			"CREATE TABLE d.types (i int, f float, s string, b bytes, d date, m timestamp, z timestamp with time zone, n interval, o bool, e decimal)",
 			[]preparedExecTest{
 				base,
 			},
 		},
 		{
-			"INSERT INTO d.types VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)",
+			"INSERT INTO d.types VALUES ($1, $2, $3, $4, $5, $6, $7::timestamptz, $8, $9, $10)",
```
Comment (dt) [inline sql/pgwire_test.go]: I'm a little bummed about this cast... for some reason I thought we were able to resolve the desired type here without it, but I guess we're missing the info when we type the placeholders?

Comment (nvanbenschoten) [inline sql/pgwire_test.go]: I'm surprised you need the cast. What error are you getting without it?

**Comment by tamird** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[sql/pgwire_test.go, line 783 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7642#-KM-brbPS19nnxnJQ21I:-KM-brbPS19nnxnJQ21J:246173814) ([raw file](https://github.com/cockroachdb/cockroach/blob/ef5484df855bc19711dedd95185baf19302dd0d9/sql/pgwire_test.go#L783)):*
> ```Go
> 					time.Time{}, // date
> 					time.Time{}, // timestamp
> 					time.Time{}, // timestamp
> ```

timestampz?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7642)*

**Comment by mjibson** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[sql/pgwire/types.go, line 389 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7642#-KM-jGX_sd-tjKQ1Bg8E:-KM-jGXaRKaodHc5Ud66:1764955675) ([raw file](https://github.com/cockroachdb/cockroach/blob/ef5484df855bc19711dedd95185baf19302dd0d9/sql/pgwire/types.go#L389)):*
> ```Go
> 		reflect.TypeOf(parser.TypeDecimal):     oid.T_numeric,
> 		reflect.TypeOf(parser.TypeString):      oid.T_text,
> 		reflect.TypeOf(parser.TypeTimestamp):   oid.T_timestamp,
> ```

Do we know why a TypeTimestamp was previously mapped to a T_timestamptz instead of a T_timestamp? Bug or feature?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7642)*

File: sql/pgwire_test.go (hunk 1 above)
Comment (dt) [inline sql/pgwire_test.go]: ```	pgwire_test.go:817: INSERT INTO d.types VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10): [0 0  [] 0001-01-01 00:00:00 +0000 UTC 0001-01-01 00:00:00 +0000 UTC 0001-01-01 00:00:00 +0000 UTC 1h0m0s true 0.0]: unexpected error: pq: value type timestamp doesn't match type TIMESTAMPTZ of column "z"```

**Comment by dt** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[sql/pgwire_test.go, line 774 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7642#-KM-IKKFy-9R8m2c5XyG-r1-774:-KM-mCyuwVoVVNkO3vVS:397049048) ([raw file](https://github.com/cockroachdb/cockroach/blob/ef5484df855bc19711dedd95185baf19302dd0d9/sql/pgwire_test.go#L774)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary>
> Do you mind opening a bug and assigning me?
</details>
Done: #7647

---

*[sql/pgwire_test.go, line 783 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7642#-KM-brbPS19nnxnJQ21I:-KM-lLPDHBVtwFuuJy-P:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/ef5484df855bc19711dedd95185baf19302dd0d9/sql/pgwire_test.go#L783)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> timestampz?
</details>
Done.

---

*[sql/pgwire/types.go, line 389 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7642#-KM-jGX_sd-tjKQ1Bg8E:-KM-kzF993CKhVP3kRnQ:-890038609) ([raw file](https://github.com/cockroachdb/cockroach/blob/ef5484df855bc19711dedd95185baf19302dd0d9/sql/pgwire/types.go#L389)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary>
> Do we know why a TypeTimestamp was previously mapped to a T_timestamptz instead of a T_timestamp? Bug or feature?
</details>
I think it might have predated sending the correct string format -- we were always sending offsets originally, which is only valid with the `tz` format code -- now that we only send offsets for non-tz, i think we can send the matching format code.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7642)*

**Comment by mjibson** [comment]
:lgtm:

---

Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7642#-:-KM-tiEgMomQYbtaLV7k:1417088721)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7642)*

**Pull request merged**